### PR TITLE
Finish otel upgrade

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

This partially resolves some comments from the otel upgrade. This specifically does not handle the otel kafka conversion, but that will be coming.

I did not update the changelog since functionally nothing changed.

- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.exporter.otlphttp.md [hunk 1]
```diff
@@ -177,4 +179,4 @@ Connecting some components may not be sensible or components may require further
 Refer to the linked documentation for more details.
 {{< /admonition >}}
 
-<!-- END GENERATED COMPATIBLE COMPONENTS -->
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/otelcol.exporter.otlphttp.md:180]: I am very confused why this is flagged has a change

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/otelcol.exporter.otlphttp.md:180]: It's tagged in the previous file as well. No idea. I don't see a difference.... maybe an empty line was removed? Git does odd things when empty lines are removed.

File: docs/sources/shared/flow/reference/components/extract-field-block.md [hunk 2]
```diff
@@ -12,13 +12,13 @@ headless: true
...
-------------|----------|----------------------------------------------------------------------------------------|---------|---------
-`from`      | `string` | The source of the labels or annotations. Allowed values are `pod` and `namespace`.     | `pod`   | no
-`key_regex` | `string` | A regular expression used to extract a key that matches the regular expression.        | `""`    | no
-`key`       | `string` | The annotation or label name. This key must exactly match an annotation or label name. | `""`    | no
-`regex`     | `string` | An optional field used to extract a sub-string from a complex field value.             | `""`    | no
-`tag_name`  | `string` | The name of the resource attribute added to logs, metrics, or spans.                   | `""`    | no
+Name        | Type     | Description                                                                                   | Default | Required
+------------|----------|-----------------------------------------------------------------------------------------------|---------|---------
+`from`      | `string` | The source of the labels or annotations. Allowed values are `pod`, `namespace` and `node`.    | `pod`   | no
```
Comment (clayton-cornell) [inline docs/sources/shared/flow/reference/components/extract-field-block.md:17]:
```suggestion
`from`      | `string` | The source of the labels or annotations. Allowed values are `pod`, `namespace`, and `node`.    | `pod`   | no
```

**Review by wildum — APPROVED** [review]
lgtm, thanks for following up on this

**Pull request merged**